src/machines/activity_state_machine.rs

```diff
@@ -305,6 +305,20 @@ impl TryFrom<CommandType> for ActivityMachineEvents {
 
 impl Cancellable for ActivityMachine {
     fn cancel(&mut self) -> Result<Vec<MachineResponse>, MachineError<Self::Error>> {
+        if matches!(
+            self.state(),
+            ActivityMachineState::Completed(_)
+                | ActivityMachineState::Canceled(_)
```

Comment:
Not sure we should allow double cancel.
At least in TS we don't issue cancellations twice.

Reply:
I think we might as well. Either we fail a situation that was otherwise recoverable, if misbehaving, or we allow it to proceed. This way if it does happen we keep operating which I think is probably better than not.

Reply:
The commands sent to core show up in the logs so we still have an avenue to detect that sort of misbehavior.